src/libmongoc/src/mongoc/mongoc-counters-private.h

```diff
@@ -101,6 +101,16 @@ _mongoc_sched_getcpu (void)
 }
 #elif defined(MONGOC_HAVE_SCHED_GETCPU)
 #define _mongoc_sched_getcpu sched_getcpu
+#elif defined(__APPLE__) && defined(__aarch64__)
+static BSON_INLINE unsigned
+_mongoc_sched_getcpu (void)
+{
+   uintptr_t tls;
+   unsigned core_id;
+   __asm__ volatile("mrs %x0, tpidrro_el0" : "=r"(tls));
```

Comment:
https://developer.arm.com/documentation/den0024/a/The-Memory-Management-Unit/Context-switching?lang=en notes:

> Additionally, the ARMv8-A Architecture provides Thread ID registers for use by operating system software. These have no hardware significance and are typically used by threading libraries as a base pointer to per-thread data.

Would reading `MPIDR_EL1` be more appropriate? References:
- https://developer.arm.com/documentation/den0024/a/Multi-core-processors/Multi-processing-systems/Determining-which-core-the-code-is-running-on?lang=en
- https://community.arm.com/support-forums/f/architectures-and-processors-forum/6978/armv8-a-instruction-for-getting-cpu-number

Reply:
What you re referring to is considered illegal in non privileged mode.

Reply:
I would prefer use of [ARM C Language Extensions](https://gcc.gnu.org/onlinedocs/gcc/ARM-C-Language-Extensions-_0028ACLE_0029.html) (test for `__ARM_ACLE` -> `#include <arm_acle.h>` -> `__arm_rsr64()` compiler intrinsic?) over inline ASM, but I believe the code should work fine as proposed.